lib/replset.js

```diff
@@ -158,6 +182,15 @@ class ReplSet extends EventEmitter {
         // Get the version information
         var result = yield self.discover();
 
+        // Set protocol version for replicaset to be given value in options or
+        // default to version 1 if using MongoDB 3.2 or above
+        // Protocol version 1 is supported since 3.2+ and required for 4.0+
+        var protocolVersion = self.protocolVersion;
```

Comment:
while we're at it: generally prefer `let` and `const` (so called lexically scoped variables) to `var`. Its not something that we go out of our way to change, but instead change when working in a piece of code or writing new code.

Reply:
👍